roles/lib_openshift/library/docker_creds.py

```diff
@@ -0,0 +1,207 @@
+        required: true
+    password:
+        description:
+            - This is the password to authenticate to the registry with.
+        required: true
+
+author:
+    - "Michael Gugino <mgugino@redhat.com>"
+'''
+
+EXAMPLES = '''
+# Pass in a message
+- name: Place credentials in file
+  docker_creds:
+    path: /root/.docker
+    registry: registry.example.com:443
+    username: myuser
+    password: mypassword
+'''
+
+import base64
+import json
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+
+def check_dest_dir_exists(module, dest):
+    '''Check if dest dir is present and is a directory'''
+    dir_exists = os.path.exists(dest)
+    if dir_exists:
+        if not os.path.isdir(dest):
+            msg = "{} exists but is not a directory".format(dest)
+            result = {'failed': True,
+                    'changed': False,
+                    'msg': msg,
+                    'state': 'unknown'}
+            module.fail_json(**result)
+        else:
+            return 1
+    else:
+        return 0
+
+def create_dest_dir(module, dest):
+    try:
+        os.makedirs(dest, mode=0o700)
+    except Exception as e:
+        result = {'failed': True,
+                'changed': False,
+                'msg': str(e),
+                'state': 'unknown'}
+        module.fail_json(**result)
+
+def load_config_file(module, dest):
+    '''load the config.json in directory dest'''
+    conf_file_path = os.path.join(dest, 'config.json')
+    if os.path.exists(conf_file_path):
+        # Try to open the file
+        try:
+            with open(conf_file_path) as f:
+                data = f.read()
+        # Couldn't open the file, exit.
+        except Exception as e:
+            result = {'failed': True,
+                    'changed': False,
+                    'msg': str(e),
+                    'state': 'unknown'}
+            module.fail_json(**result)
+        try:
+            j = json.loads(data)
+        # Invalid json data.
+        except Exception as e:
+            result = {'failed': True,
+                    'changed': False,
+                    'msg': str(e),
+                    'state': 'unknown'}
+            module.fail_json(**result)
+        return j
+    else:
+        return {}
+
+def update_config(docker_config, registry, username, password):
+    '''Add our registry auth credentials into docker_config dict'''
+
+    # Add anything that might be missing in our dictionary
+    if 'auths' not in docker_config:
+        docker_config['auths'] = {}
+    if registry not in docker_config['auths']:
+        docker_config['auths'][registry] = {}
+
+    # base64 encode our username:password string
+    encoded_data = base64.b64encode('{}:{}'.format(username, password))
+
+    # check if the same value is already present for idempotency.
+    if 'auth' in docker_config['auths'][registry]:
+        existing_data = docker_config['auths'][registry]['auth']
```

Comment:
Prefer not to create an extra variable only to compare.  Suggest using the `docker_config['auths'][registry]['auth']` and compare it on the next line.